.travis.yml

```diff
@@ -2,11 +2,13 @@ language: rust
 rust:
   - stable
   - nightly
-  - 1.22.0  # project wide min version
+  - 1.36.0  # project wide min version
```

Comment:
I'm seeing `1.29` on the main `rust-bitcoin` repository.

Reply:
The PR description explains why i chose 1.36. I'm afraid this solution isn't supported on earlier versions.

Reply:
There are workarounds for conditional compilation against the version of rustc ([rustversion](https://github.com/dtolnay/rustversion), [rustc-version-rs](https://github.com/Kimundi/rustc-version-rs)) perhaps it is worth considering if backwards compatibility is necessary?

Reply:
Thanks for those links @gregdhill, but I think one of the strengths of this library is having zero dependencies.

It might be better do maintain a fork of the repo with this work and merge once the `rust-bitcoin` project moves past Rust 1.36.

Reply:
While raising the *general* msrv is a non-starter I don't think this is necessary here. It appears the incompatibility comes from using the `alloc` crate which only happens in no-std mode. That means we should still test the against rust 1.22.0 in std mode but against 1.36.0 in no_std mode. I think that would be a reasonable compromise. `rust-bitcoin` can still depend on the std version compatible with 1.22.0 and e.g. embedded devs that have to use more recent versions anyway can use no_std.

That being said, I don't need it currently, so won't put work in beyond reviewing/approving. What do you think @clarkmoody, would this be acceptable? Also: please don't merge in master into your branch, rather rebase.

Reply:
@sgeisler That proposal sounds good. Would this be reflected in an update to `.travis.yml`?

Reply:
Depends on if you want to run tests in no-std mode. If there was a feature that controlled std-availability, lime most crates do it, we could just add a dimension to our test matrix afaik to test everything in no-std mode too for stable and nightly (just not 1.22/1.29).